### PR TITLE
[Enhancement] Add BASE_TABLE_REFRESH_VERSION_TIMES column to information_schema.materialized_views

### PR DESCRIPTION
**Pull request by Youngwb**
## Why I'm doing:

`information_schema.materialized_views` and `SHOW MATERIALIZED VIEWS` already expose `LAST_REFRESH_TIME`, but only as a single maximum over all base tables. When a materialized view has multiple base tables, operators cannot see which base table the reported data version time comes from, nor the per-table breakdown. This makes it hard to reason about staleness on a per-base-table basis — including for cloud-native IVM materialized views.

## What I'm doing:

Add a new column `BASE_TABLE_REFRESH_VERSION_TIMES` (varchar) to `information_schema.materialized_views` and `SHOW MATERIALIZED VIEWS`. It is a JSON object mapping each base table's `catalog.database.table` name to the latest data version time observed for it — the per-table detail behind `LAST_REFRESH_TIME` (their single maximum). `{}` when no base table has a recorded time.

It covers all base table kinds:
- **external/data lake** tables report their partition source modified time (from `getBaseTableInfoVisibleVersionMap`);
- **OLAP/internal** tables report their visible-version commit time (from `getBaseTableVisibleVersionMap`, resolving the table id to `catalog.database.table`).

Both PCT refresh and cloud-native IVM populate these version maps through `MVVersionManager` (IVM's refresh path calls `updatePCTMeta` → `updateMetaForOlapTable`/`updateMetaForExternalTable`), so IVM-refreshed materialized views are covered too.

The column is wired through every existing column surface: the `information_schema` table definition, the SHOW result header, the `SHOW ... WHERE` → SELECT rewrite column list, `TMaterializedViewStatus` thrift field 39, `ShowMaterializedViewStatus` (thrift + result row), and the BE schema scanner.

The stored modified time is intentionally left in the connector's native unit (iceberg microseconds, hive/jdbc/odps seconds, paimon and OLAP milliseconds), because partition-change detection and staleness compare it precisely against the live value; converting it at storage would lose sub-millisecond precision and make iceberg partitions look perpetually changed. This column therefore normalizes to epoch millis **only at render time** via `TimeUtils.normalizeToEpochMillis`, inferring the unit by magnitude (boundary matching `DefaultTraits.normalizeIcebergModifiedTimeToMicros`). Without this, raw iceberg microseconds format as year 57221 and hive seconds as ~1970.

Validated on a dev cluster (FE+BE deployed): an OLAP-backed MV → `{"default_catalog.db.t":"<datetime>"}`; a single iceberg MV → `{"<catalog>.<db>.<table>":"<datetime>"}`; a two-iceberg-table join MV → both keys present (verified order-independently via `get_json_string`). End-to-end SQL cases cover the OLAP and single/multi external paths; unit tests cover OLAP, external (microsecond render), and the empty/null cases.

Fixes #issue: N/A — observability enhancement, no tracked issue.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function
  - [ ] This pr needs auto generate documentation
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 4.1
  - [ ] 4.0
  - [ ] 3.5

### CODE REVIEW

**Comment by CelerData-Reviewer** [comment]
@codex review

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtToSelectStmtConverter.java [hunk 1]
```diff
@@ -96,7 +96,8 @@ public final class ShowStmtToSelectStmtConverter {
             "refresh_policy",
             "resource_group",
             "query_rewrite_status_reason",
-            "last_freshness_confirmed_at"
+            "last_freshness_confirmed_at",
+            "base_table_refresh_version_times"
```
[the remark this answers is not in the document]
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtToSelectStmtConverter.java:100]: Fixed in `6bfeb8a` — updated the expected rewrite string in `ShowMaterializedViewTest.testNormal` to append `information_schema.materialized_views.base_table_refresh_version_times AS base_table_refresh_version_times` before the `FROM` clause, matching the `AstToStringBuilder` output verbatim. (This Codex review ran on the earlier commit `03d210e5`, prior to the fix.)

**Comment by Youngwb** [comment]
@codex review

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java [hunk 2]
```diff
@@ -981,6 +984,29 @@ public void setRefreshScheme(MvRefreshScheme refreshScheme) {
...
+        Map<String, String> result = new LinkedHashMap<>();
+        for (Map.Entry<BaseTableInfo, Map<String, BasePartitionInfo>> entry :
+                scheme.getAsyncRefreshContext().getBaseTableInfoVisibleVersionMap().entrySet()) {
+            long maxRefreshTime = entry.getValue().values().stream()
+                    .mapToLong(BasePartitionInfo::getLastRefreshTime)
+                    .filter(t -> t > 0)
+                    .max()
+                    .orElse(0L);
+            if (maxRefreshTime > 0) {
+                result.put(entry.getKey().getReadableString(), TimeUtils.longToTimeString(maxRefreshTime));
```
[the remark this answers is not in the document]
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java]: Good catch — fixed in `6ba639b`. The new column now normalizes `lastRefreshTime` to epoch millis by magnitude at read time (`normalizeRefreshTimeToMillis`: s/ms/µs, boundaries matching `DefaultTraits.normalizeIcebergModifiedTimeToMicros`), so legacy metadata persisted before `fromExternalTable` normalized units renders a real datetime immediately after upgrade — no re-refresh needed. Verified on a dev cluster (iceberg MV → real datetime) and with a unit test asserting a raw-microsecond value renders correctly. Note: `LAST_REFRESH_TIME` (ShowMaterializedViewStatus) shares the same pre-existing legacy-unit behavior but is untouched by this PR; tracking it as a separate follow-up to avoid changing a long-standing column here.

**Comment by Youngwb** [comment]
@codex review

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java [hunk 3]
```diff
@@ -267,7 +267,10 @@ public BasePartitionInfo(long id, long version, long lastRefreshTime) {
         }
 
         public static BasePartitionInfo fromExternalTable(com.starrocks.connector.PartitionInfo info) {
-            return new BasePartitionInfo(-1, info.getVersion(), info.getModifiedTime());
+            // Normalize lastRefreshTime to epoch millis; connectors report modified time in different units
+            // (iceberg µs, hive/jdbc/odps s, paimon ms). version is left as-is (used by change detection).
+            return new BasePartitionInfo(-1, info.getVersion(),
+                    info.getModifiedTimeUnit().toMillis(info.getModifiedTime()));
```
[the remark this answers is not in the document]
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java]: Correct, and thanks — this is exactly the regression. Reverted in `5efa3fa`: `fromExternalTable` no longer converts units, so the persisted `BasePartitionInfo.lastRefreshTime` stays in the connector's native unit (iceberg µs, hive/jdbc/odps s, paimon ms). That keeps `isStalenessSatisfied()` / `maxPartitionRefreshTs()` consistent (your Hive/JDBC case), and also avoids the µs→ms→µs precision round-trip that was making iceberg partition-change detection see every partition as changed (FE UT caught ~36 iceberg/hive failures on `6ba639b`). The new column is now purely additive and does the unit normalization only at render time (`normalizeRefreshTimeToMillis`, magnitude-based), so it shows a real datetime without touching any value used by change detection or staleness. The production diff vs main is now just the new method + that read-time helper.

**Comment by Youngwb** [comment]
@codex review

**Comment by Youngwb** [comment]
@codex review

**Review by HangyuanLiu — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]
